### PR TITLE
[KernelGen][Nvidia] Add _batch_norm_impl_index operator with Triton kernel

### PR DESCRIPTION
**Pull request by YufanPeter**
### Summary

  Adds a Triton implementation of `_batch_norm_impl_index` (`torch.ops.aten._batch_norm_impl_index`), the
  index-dispatching batch-norm wrapper. The aten schema returns a 5-tuple `(output, save_mean, save_invstd,
  reserve, impl_index)`.

  The forward kernel reuses the proven attorch-derived `batch_norm_forward_kernel` (the same kernel backing
  `flag_gems.batch_norm` and `_native_batch_norm_legit_functional`): a 1D grid over channels, each program
  running a serial Welford reduction over its `(batch × spatial)` slice, with the normalization applied in
  the same launch. Schema-specific handling:

  - `save_invstd` returns `inv_std = 1/sqrt(var + eps)`, matching PyTorch's convention (PyTorch's
  "save_var" is in fact the inverse std, not the variance).
  - `reserve` is an empty `uint8` tensor and `impl_index` is `0`, because this Triton fallback uses neither
  cuDNN nor MIOpen (the torch reference also returns `impl_index=0` on this GPU, so the values are
  consistent).
  - `cudnn_enabled` is accepted for schema compatibility but intentionally ignored, since the path never
  dispatches to cuDNN.

  ### Testing

  - Parametrized over 3D/4D/5D shapes, affine (`weight`/`bias` present) True/False, and dtypes `float16 /
  float32 / bfloat16`
  - Coverage: 2D `(16, 3)`, 3D `(32, 32, 32)`, 4D `(8, 32, 224, 224)` and `(2050, 16, 32, 32)`, and 5D `(8,
  16, 3, 224, 224)`; running-stats update path exercised with `training=True`
  - All five returns validated against the fp64-upcast reference via `to_reference` + `gems_assert_close`
  (output, save_mean, save_invstd, running_mean, running_var)
  - **All 30 accuracy cases pass**

  ### Performance

  Test command: `pytest benchmark/test__batch_norm_impl_index.py --level core` (NVIDIA H20)

  | dtype | shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
  |-------|-------|--------------------|-------------------|---------|
  | float16 | [4, 16, 64, 4] | 0.012 | 0.006 | 1.886 |
  | float16 | [16, 16, 8, 48] | 0.020 | 0.008 | 2.472 |
  | float16 | [16, 16, 8, 88] | 0.025 | 0.010 | 2.434 |
  | float16 | [16, 16, 128] | 0.014 | 0.007 | 2.098 |
  | float16 | [20, 6, 65536] | 1.070 | 0.335 | 3.196 |
  | float32 | [4, 16, 64, 4] | 0.007 | 0.006 | 1.170 |
  | float32 | [16, 16, 8, 48] | 0.009 | 0.009 | 1.032 |
  | float32 | [16, 16, 8, 88] | 0.012 | 0.011 | 1.073 |
  | float32 | [16, 16, 128] | 0.008 | 0.007 | 1.165 |
  | float32 | [20, 6, 65536] | 0.592 | 0.535 | 1.107 |
  | bfloat16 | [4, 16, 64, 4] | 0.012 | 0.006 | 1.919 |
  | bfloat16 | [16, 16, 8, 48] | 0.020 | 0.008 | 2.537 |
  | bfloat16 | [16, 16, 8, 88] | 0.026 | 0.010 | 2.523 |
  | bfloat16 | [16, 16, 128] | 0.012 | 0.007 | 1.878 |
  | bfloat16 | [20, 6, 65536] | 1.075 | 0.340 | 3.163 |

  #### Arithmetic Mean Speedup

  | dtype | Mean Speedup |
  |-------|--------------|
  | float16 | 2.417 |
  | float32 | 1.109 |
  | bfloat16 | 2.404 |
  | **Overall** | **1.977x** |

  ### Multi-backend Testing

  | Backend | Accuracy Test | Speedup (mean) | Notes |
  |---|---|---|---|
  | Nvidia (H20-3e) | PASS (30 cases) | 1.977 (overall) | Primary (fp16 / fp32 / bf16) |

  ### Files Changed

  - `src/flag_gems/ops/_batch_norm_impl_index.py`: Triton kernel implementation (channel-parallel Welford
  BN forward + normalization)
  - `tests/test__batch_norm_impl_index.py`: Accuracy test
  - `benchmark/test__batch_norm_impl_index.py`: Performance benchmark
  - `src/flag_gems/ops/__init__.py`: Register import and `__all__`
  - `src/flag_gems/__init__.py`: Register to config
  - `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: src/flag_gems/ops/__init__.py [hunk 1]
```diff
@@ -758,6 +756,7 @@
     "_add_relu",
     "_amp_foreach_non_finite_check_and_unscale_",
     "_assert_async",
+    "_batch_norm_impl_index",
```
Comment (Caeruleann) [inline src/flag_gems/ops/__init__.py:761]:
```suggestion
    "batch_norm_impl_index",
```

File: src/flag_gems/ops/__init__.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
 from flag_gems.ops._amp_foreach_non_finite_check_and_unscale_ import (
     _amp_foreach_non_finite_check_and_unscale_,
 )
+from flag_gems.ops._batch_norm_impl_index import _batch_norm_impl_index
```
Comment (Caeruleann) [inline src/flag_gems/ops/__init__.py]:
```suggestion
from flag_gems.ops._batch_norm_impl_index import batch_norm_impl_index
```

**Review by Caeruleann — DISMISSED** [review]

**Review by Caeruleann — APPROVED** [review]
